### PR TITLE
Add V2 script context equivalence test shell scripts

### PR DESCRIPTION
**Pull request by Jimbo4350**
This pr adds two plutus scripts with accompanying shell scripts:
`scripts/babbage/script-context-equivalance-test.sh`
`scripts/babbage/script-mint-context-equivalance-test.sh`

The plutus scripts test various fields in the `ScriptContext` for equivalence in our redeemer that is created by the `create-script-context` executable (from `plutus-example` in `plutus-apps`). The shell script succeed depending on if the fields in the `ScriptContext` are actually equivalent to what is in our redeemer.

### CODE REVIEW

File: scripts/babbage/script-context-equivalance-test.sh [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+#Get read only reference input
+$CARDANO_CLI query utxo --address "$readonlyaddress" --cardano-mode \
+  --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/read-only-ref-input-utxo.json
+readonlyrefinput=$(jq -r 'keys[0]' $WORK/read-only-ref-input-utxo.json)
+
+
+
+# We need to generate a dummy redeemer in order to create a txbody from which we can generate
+# a tx and then derive the correct redeemer. TODO: Why? The cli demands you specify a redeemer (double check this)
```
Comment (james-iohk) [inline scripts/babbage/script-context-equivalance-test.sh:107]: Does this todo need addressing? I think we know the answer. (same for minting bash script)

File: scripts/babbage/script-mint-context-equivalance-test.sh [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+echo "Socket path: $CARDANO_NODE_SOCKET_PATH"
+echo "Socket path: $(pwd)"
+
+ls -al "$CARDANO_NODE_SOCKET_PATH"
+
+# NB: This plutus script uses a "typed" redeemer and "typed" datum.
+plutusscriptinuse="$BASE/scripts/plutus/scripts/v2/minting-context-equivalance-test.plutus"
+policyid=$(cardano-cli transaction policyid --script-file $plutusscriptinuse)
+# This datum hash is the hash of the typed 42
```
Comment (james-iohk) [inline scripts/babbage/script-mint-context-equivalance-test.sh:27]: There is no datumhash for minting

Comment (Jimbo4350) [reply scripts/babbage/script-mint-context-equivalance-test.sh:27]: copy/paste error

File: scripts/babbage/script-context-equivalance-test.sh (hunk 1 above)
Comment (Jimbo4350) [reply scripts/babbage/script-context-equivalance-test.sh:107]: Nop, I'll remove the TODO

File: scripts/babbage/script-context-equivalance-test.sh [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+export CARDANO_NODE_SOCKET_PATH="${CARDANO_NODE_SOCKET_PATH:-example/main.sock}"
+export TESTNET_MAGIC="${TESTNET_MAGIC:-42}"
+export UTXO_VKEY="${UTXO_VKEY:-example/utxo-keys/utxo1.vkey}"
+export UTXO_SKEY="${UTXO_SKEY:-example/utxo-keys/utxo1.skey}"
+export RESULT_FILE="${RESULT_FILE:-$WORK/result.out}"
+
+mkdir -p "$WORK"
+
+echo "Socket path: $CARDANO_NODE_SOCKET_PATH"
+echo "Socket path: $(pwd)"
```
Comment (newhoggy) [inline scripts/babbage/script-context-equivalance-test.sh:20]: This echo doesn't seem right.

File: scripts/babbage/script-context-equivalance-test.sh [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  --stake-script-file "$certifyingscript" \
+  --out-file "$WORK/script.regcert"
+
+# Step 1: Create a tx output with a datum hash at the script address. In order for a tx output to be locked
+# by a plutus script, it must have a datahash. We also need collateral tx inputs so we split the utxo
+# in order to accommodate this.
+
+
+plutusscriptaddr=$($CARDANO_CLI address build --payment-script-file "$plutusscriptinuse"  --testnet-magic "$TESTNET_MAGIC")
+readonlyaddress=addr_test1vz3t3f2kgy2re66tnhgxc4t8jgylw2cqfnxdwlrq9agfmtstxxkm5
```
Comment (newhoggy) [inline scripts/babbage/script-context-equivalance-test.sh:43]: What does this address correspond to?

File: scripts/babbage/script-context-equivalance-test.sh [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+$CARDANO_CLI query utxo --address $utxoaddr --cardano-mode --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/utxo-2.json
+cat $WORK/utxo-2.json
+txinCollateral=$(jq -r 'keys[0]' $WORK/utxo-2.json)
+
+
+dummyaddress=addr_test1vpqgspvmh6m2m5pwangvdg499srfzre2dd96qq57nlnw6yctpasy4
+
+lovelaceatplutusscriptaddr=$(jq -r ".[\"$plutusutxotxin\"].value.lovelace" $WORK/plutusutxo.json)
+
+#Get read only reference input
```
Comment (newhoggy) [inline scripts/babbage/script-context-equivalance-test.sh:99]: Space before `Get`.

File: scripts/babbage/script-context-equivalance-test.sh [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+$CARDANO_CLI query utxo --address "$readonlyaddress" --cardano-mode \
+  --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/read-only-ref-input-utxo.json
+readonlyrefinput=$(jq -r 'keys[0]' $WORK/read-only-ref-input-utxo.json)
+
+
+
+# We need to generate a dummy redeemer in order to create a txbody from which we can generate
+# a tx and then derive the correct redeemer.
```
Comment (newhoggy) [inline scripts/babbage/script-context-equivalance-test.sh:107]: Please add an explanation for this need.

File: scripts/babbage/script-mint-context-equivalance-test.sh [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+export CARDANO_NODE_SOCKET_PATH="${CARDANO_NODE_SOCKET_PATH:-example/main.sock}"
+export TESTNET_MAGIC="${TESTNET_MAGIC:-42}"
+export UTXO_VKEY="${UTXO_VKEY:-example/utxo-keys/utxo1.vkey}"
+export UTXO_SKEY="${UTXO_SKEY:-example/utxo-keys/utxo1.skey}"
+export RESULT_FILE="${RESULT_FILE:-$WORK/result.out}"
+
+mkdir -p "$WORK"
+
+echo "Socket path: $CARDANO_NODE_SOCKET_PATH"
+echo "Socket path: $(pwd)"
```
Comment (newhoggy) [inline scripts/babbage/script-mint-context-equivalance-test.sh:20]: This echo seems wrong.

File: scripts/babbage/script-mint-context-equivalance-test.sh [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+cat $WORK/utxo-2.json
+txinCollateral=$(jq -r 'keys[0]' $WORK/utxo-2.json)
+txinfunding1=$(jq -r 'keys[1]' $WORK/utxo-2.json)
+
+
+dummyaddress=addr_test1vpqgspvmh6m2m5pwangvdg499srfzre2dd96qq57nlnw6yctpasy4
+
+
+# We need to generate a dummy redeemer in order to create a txbody from which we can generate
+# a tx and then derive the correct redeemer.
```
Comment (newhoggy) [inline scripts/babbage/script-mint-context-equivalance-test.sh:74]: Please add an explanation for this need.

**Review by newhoggy — APPROVED** [review]

File: scripts/babbage/script-context-equivalance-test.sh (hunk 4 above)
Comment (Jimbo4350) [reply scripts/babbage/script-context-equivalance-test.sh:43]: The input at this address will be used as a reference input. I've added a comment

**Pull request merged**